### PR TITLE
SimJTAG: add `--remote-jtag-port` to specify remote bitbang port

### PR DESCRIPTION
**Pull request by Tang-Haojin**

### CODE REVIEW

File: src/test/csrc/common/SimJTAG.cpp [hunk 1]
```diff
@@ -19,7 +20,7 @@ extern "C" int jtag_tick(unsigned char *jtag_TCK, unsigned char *jtag_TMS, unsig
     return 0;
   if (!jtag) {
     // TODO: Pass in real port number
```
Comment (poemonsense) [inline src/test/csrc/common/SimJTAG.cpp:22]: Let's remove this TODO and add a separate function `jtag_init`. This function can be reused here in SimJTAG.cpp as well as emu.cpp.

```c++
void jtag_init() {
  jtag = new remote_bitbang_t(remote_jtag_port);
}

// in SimJTAG.cpp
if(!jtag) {
  jtag_init();
};

// in emu.cpp
if (args.enable_jtag) {
  remote_jtag_port = args.remote_jtag_port;
  jtag_init();
}
```

File: src/test/csrc/verilator/emu.cpp [hunk 2]
```diff
@@ -134,6 +135,7 @@ inline EmuArgs parse_args(int argc, const char *argv[]) {
     { "no-diff",           0, NULL,  0  },
     { "enable-fork",       0, NULL,  0  },
     { "enable-jtag",       0, NULL,  0  },
+    { "remote-jtag-port",  1, NULL,  0  },
```
Comment (poemonsense) [inline src/test/csrc/verilator/emu.cpp:138]: In the `--help` message, it is better to put it after `--enable-jtag`. 

However, I think we do not need to put this option here. We can append this argument at the end of the argument list (struct). Then we do not need to change the indices for prior arguments, and there will be less lines of code changes. This order change will not affect the functionalities.

**Review by poemonsense — COMMENTED** [review]
Propose some style changes

File: src/test/csrc/verilator/emu.h [hunk 3]
```diff
@@ -37,6 +37,7 @@
 #endif
 
 struct EmuArgs {
+  uint16_t remote_jtag_port = 23334;
```
Comment (poemonsense) [inline src/test/csrc/verilator/emu.h:40]: We can leave the default value only at SimJTAG.cpp.

We also do not need to add the argument in EmuArgs. We can set remote_jtag_port when parsing the arguments.

```c++
case SOME_NUMBER: remote_jtag_port  = atoll_strict(optarg, "remote-jtag-port"); continue;
```

File: src/test/csrc/verilator/emu.cpp (hunk 2 above)
Comment (Tang-Haojin) [reply src/test/csrc/verilator/emu.cpp]: Fixed.

File: src/test/csrc/verilator/emu.h (hunk 3 above)
Comment (Tang-Haojin) [reply src/test/csrc/verilator/emu.h]: Fixed. Also remove redundant `enable_jtag`.

File: src/test/csrc/common/SimJTAG.cpp (hunk 1 above)
Comment (Tang-Haojin) [reply src/test/csrc/common/SimJTAG.cpp]: Fixed.

**Review by poemonsense — APPROVED** [review]

**Pull request merged**